packages/corelib/src/dataModel/ShowStyleVariant.ts

```diff
@@ -4,6 +4,9 @@ import { ShowStyleVariantId, ShowStyleBaseId } from './Ids'
 
 export interface DBShowStyleVariant extends ProtectedStringProperties<IBlueprintShowStyleVariant, '_id'> {
 	_id: ShowStyleVariantId
+
+	_rank?: number
```

Comment:
I don't think this should be an optional property, I would prefer it to be required to be a number.

This will require a little more work (a migration) to ensure it is set when appropriate, and determining a value when inserting one, but I think that is better and more predictable than everywhere which uses the rank having to figure out what to do with the ones with an undefined rank. (should they go at the beginning/end/randomly?)

Reply:
A very good point, i will focus on the other review comments today and give this an attempt tomorrow.